java/src/main/native/src/RmmJni.cpp

```diff
@@ -395,15 +395,17 @@ class java_debug_event_handler_memory_resource final : public java_event_handler
   }
 };
 
-inline auto& prior_cuio_host_mr()
+inline auto& prior_cudf_pinned_mr()
```

Comment:
Should not `inline` since this function contains static variable.
```suggestion
auto& prior_cudf_pinned_mr()
```

Reply:
ok this is me not knowing what I am doing. If `inline` means we can have multiple copies of `prior_cudf_pinned_mr` that's bad. We don't want to have an inconsistent view of the "prior" mr.

Today we use this at the end of the job, so we could see crashes if we reset to a an mr we have closed if we do get multiple versions of the method.

Reply:
according to stackoverflow the compile should take care of this special case:

```
A static local variable in an extern inline function always refers to the same object.

7.1.2/4 - C++98/C++14 (n3797)
```

Reply:
@ttnghia are you afraid of a critical bug here or should we file a follow on to fix? I don't think we should hold this PR if it's unrelated.

Reply:
Looked into this a bit - inlined functions still have a single instance of the local static objects:
> Function-local static objects in all function definitions are shared across all translation units (they all refer to the same object defined in one translation unit).

Reply:
oops, missed the other comments before replying

Reply:
Okay, so from the docs this seems not to cause any bug. But from my view this still looks like error-prone and is a dangerous situation--who knows if the compiler has a bug and ignores this fact with inline+static? I'm fine with keeping it, but still like to have `inline` removed. This is similar to other (unrelated) situations when I see the code looks like `*ptr++` and I always want to rewrite like `(*ptr)++`, otherwise it is very cryptic and difficult to understand whether `*ptr` is executed first or `ptr++` first.

Reply:
IIUC this behavior is specified by the standard, so we can rely on it (also, inline does not guarantee inlining, and lack on inline does not guarantee non-inlining, so the use of inline does not change the potential issue as @ttnghia sees it).
However, if the intent is to have internal linkage, using an anonymous namespace is preferable :)